vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.h

```diff
@@ -36,12 +37,16 @@ class VectorData final : public Data {
   VectorData& operator=(const VectorData& vector_data);
   VectorData& operator=(VectorData&& vector_data) noexcept;
 
+  // Mathematical vector operations
   friend double operator*(const VectorData& lhs, const VectorData& rhs);
-
+  void AddElementWise(const VectorData& v_add);
+  void DivideByScalar(float scalar);
   void Normalize();
```

Comment:
Could we not decouple the math vector operators?

Reply:
These functions are meant to act in-place, rather than returning a new VectorData object. So I think it makes most sense to define them within the VectorData class?